frontend/src/lib/auth/permissions.test.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  STATUSES_ELIGIBLE_FOR_EDIT,
+  STATUSES_ELIGIBLE_FOR_REVIEW,
+} from "../../pages/destructionlist/detail/constants";
+import { User } from "../api/auth";
+import {
+  DESTRUCTION_LIST_STATUSES,
+  DestructionList,
+} from "../api/destructionLists";
+import {
+  canMarkListAsFinal,
+  canReviewDestructionList,
+  canStartDestructionList,
+  canUpdateDestructionList,
+  canViewDestructionList,
+} from "./permissions";
+
+/**
+ * Helper function to generate user objects with different permissions
+ * @param {boolean} canStartDestruction - Permission to start destruction
+ * @param {boolean} canReviewDestruction - Permission to review destruction
+ * @returns {User} - User object with specified permissions
+ */
+const generateUserWithPermissions = (
+  canStartDestruction: boolean,
+  canReviewDestruction: boolean,
+): User => ({
+  pk: 1,
+  username: "testuser",
+  firstName: "Test",
+  lastName: "User",
+  email: "testuser@example.com",
+  role: {
+    name: "Test Role",
+    canStartDestruction,
+    canReviewDestruction,
+    canViewCaseDetails: true,
+  },
+});
+
+describe("Permissions tests with all combinations", () => {
+  DESTRUCTION_LIST_STATUSES.forEach((status) => {
+    describe(`With destruction list status: ${status}`, () => {
+      let destructionList: DestructionList;
+      let user: User;
+      let anotherUser: User;
+
+      beforeEach(() => {
+        user = generateUserWithPermissions(true, true);
+        anotherUser = { ...user, pk: 2 };
+        destructionList = {
+          pk: 1,
+          assignee: user,
+          assignees: [],
+          author: user,
+          containsSensitiveInfo: false,
+          created: "2023-01-01T00:00:00Z",
+          name: "Test Destruction List",
+          status,
+          statusChanged: null,
+          uuid: "12345-abcde",
+        };
+      });
+
+      describe("canStartDestructionList", () => {
+        it("should allow a user with the canStartDestruction role", () => {
+          expect(canStartDestructionList(user)).toBe(true);
+        });
+
+        it("should not allow a user without the canStartDestruction role", () => {
+          user = generateUserWithPermissions(false, true);
+          expect(canStartDestructionList(user)).toBe(false);
+        });
+      });
+
+      describe("canReviewDestructionList", () => {
+        it("should allow a user to review if they have the role and status is eligible", () => {
+          const isEligible = STATUSES_ELIGIBLE_FOR_REVIEW.includes(status);
```

Comment:
The assertion can be "hard coded" right? We want to mock the input (set a eligible status on the list) and check if it passes. Our test should be predicable.